File: Data Structures & Algorithms/daily-temperatures/submission-1.cs
public class Solution {
    public int[] DailyTemperatures(int[] temperatures) {
        int[] res = new int[temperatures.Length];
        
        // We only store the INDEX in the stack now!
        Stack<int> stack = new Stack<int>(); 

        for (int i = 0; i < temperatures.Length; i++) {
            
            // Look at the index at the top of the stack, 
            // and use it to check the temperatures array
            while (stack.Count > 0 && temperatures[i] > temperatures[stack.Peek()]) {
                int oldIndex = stack.Pop();
                res[oldIndex] = i - oldIndex;
            }
            
            stack.Push(i);
        }
        
        return res;
    }
}

File: Data Structures & Algorithms/height-checker/submission-1.cpp
class Solution {
public:
    int heightChecker(vector<int>& heights) {
        int count[101] = {};
        for (int h : heights) {
            count[h]++;
        }

        vector<int> expected;
        for (int h = 1; h <= 100; h++) {
            int c = count[h];
            for (int i = 0; i < c; i++) {
                expected.push_back(h);
            }
        }

        int res = 0;
        for (int i = 0; i < heights.size(); i++) {
            if (heights[i] != expected[i]) {
                res++;
            }
        }

        return res;
    }
};
